src/Microsoft.AspNetCore.Server.Testing/Deployers/RemoteDeployer/RemoteDeployer.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Server.Testing
+{
+    public class RemoteDeployer : ApplicationDeployer
+    {
+        /// <summary>
+        /// Example: If the share path is '\\foo\bar', then this returns the full path to the
+        /// deployed folder. Example: '\\foo\bar\048f6c99-de3e-488a-8020-f9eb277818d9'
+        /// </summary>
+        private string _deployedFolderPathInFileShare;
+        private readonly RemoteDeploymentParameters _deploymentParameters;
+        private static readonly string _remotePSSessionHelperScript;
+        private bool _isDisposed;
+
+        static RemoteDeployer()
+        {
+            // Copy the scripts from this assembly's embedded resources to the temp path on the machine where these
+            // tests are being run
+            var embeddedFileProvider = new EmbeddedFileProvider(
```

Comment:
Static constructors are hard to debug. Can you do this with the lazy loader pattern instead? Also, you don't need to load the embedded resources unless someone uses this class